src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -868,7 +876,14 @@ void grpc_chttp2_complete_closure_step(grpc_exec_ctx *exec_ctx,
       grpc_transport_move_stats(&s->stats, s->collecting_stats);
       s->collecting_stats = NULL;
     }
-    grpc_closure_run(exec_ctx, closure, closure->error_data.error);
+    if (t->write_state == GRPC_CHTTP2_WRITE_STATE_IDLE ||
+        (closure->next_data.scratch & CLOSURE_BARRIER_CANNOT_RUN_WITH_WRITE) ==
+            0) {
```

Comment:
Can I suggest extra parens around each operand of the || operator to avoid reader confusion when mixing with bitwise ops? Also, for a bitmask isn't `!` a more common paradigm than `== 0` ?